### PR TITLE
Move aws mocks to a library and use that.

### PR DESCRIPTION
**Pull request by tomwilkie**
I want to use this for scope too /cc @jml

### CODE REVIEW

File: vendor/github.com/tomwilkie/go-mockaws/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Mocks for aws-sdk-go
+
+This package contains a series of mock types for testing against AWS' golang SDK.
+
+To use, `import github.com/tomwilkie/go-mockaws`.
```
Comment (juliusv) [inline vendor/github.com/tomwilkie/go-mockaws/README.md]: Quotes around the package name?

File: vendor/github.com/tomwilkie/go-mockaws/s3.go [hunk 2]
```diff
@@ -31,13 +18,14 @@ type mockS3Bucket struct {
 	objects map[string][]byte
 }
 
-func newMockS3() *mockS3 {
-	return &mockS3{
+// NewMockS3 returns a new MockS£
```
Comment (juliusv) [inline vendor/github.com/tomwilkie/go-mockaws/s3.go]: MockS£ -> MockS3.

**Comment by juliusv** [comment]
:+1: in general

**Comment by jml** [comment]
Yeah, I want to use this for scope too. Why under your personal Github & not weaveworks?

**Comment by tomwilkie** [comment]
> Why under your personal Github & not weaveworks?

Why not?

**Pull request merged**